fhttp/http_client.go

```diff
@@ -426,35 +451,27 @@ func NewStdClient(o *HTTPOptions) (*Client, error) {
 			if o.Resolve != "" {
 				addr = o.Resolve + addr[strings.LastIndex(addr, ":"):]
 			}
-			return (&net.Dialer{
+			conn, err := (&net.Dialer{
 				Timeout: o.HTTPReqTimeOut,
 			}).DialContext(ctx, network, addr)
+
+			req.RemoteAddr = conn.RemoteAddr().String()
```

Comment:
probably should only if err != nil or `conn` may be nil and it would crash?

and also maybe only if not already set to avoid keep calling RemoteAddr().String() {tradeoff being between getting the first address or last address picked} - but it should be rare so that part is fine (maybe a log when it does change from previous non empty value then if we keep reassigning)